app/src/androidTest/java/com/github/sdp/mediato/DatabaseTests/UserTests.java

```diff
@@ -79,73 +80,31 @@ public void followUserAddsUsernameInFollowingAndFollowers() throws ExecutionExce
         Database.addUser(user3).get(STANDARD_USER_TIMEOUT, TimeUnit.SECONDS);
 
         Database.followUser(user2.getUsername(), user3.getUsername());
-        DatabaseReference user1FollowingRef = Database.database.getReference().child(Database.USERS_PATH + user2.getUsername() + Database.FOLLOWING_PATH);
-        user1FollowingRef.addListenerForSingleValueEvent(new ValueEventListener() {
-            @Override
-            public void onDataChange(@NonNull DataSnapshot snapshot) {
-
-                Map<String, Boolean> following = (Map<String, Boolean>) snapshot.getValue();
-                assertTrue(following.get(user3.getUsername()));
-            }
-
-            @Override
-            public void onCancelled(@NonNull DatabaseError error) {
-                throw error.toException();
-            }
-        });
-
-        DatabaseReference user2FollowersRef = Database.database.getReference().child(Database.USERS_PATH + user3.getUsername() + Database.FOLLOWERS_PATH);
-        user2FollowersRef.addListenerForSingleValueEvent(new ValueEventListener() {
-            @Override
-            public void onDataChange(@NonNull DataSnapshot snapshot) {
-
-                Map<String, Boolean> followers = (Map<String, Boolean>) snapshot.getValue();
-                assertTrue(followers.get(user2.getUsername()));
-            }
-
-            @Override
-            public void onCancelled(@NonNull DatabaseError error) {
-                throw error.toException();
-            }
-        });
+        Thread.sleep(STANDARD_SLEEP_DELAY);

```

Comment:
nice way more clean

Reply:
Is it really cleaner ?  It might take more than one second to finish, and we are forced to wait 1 second. Would using futures with onTimeout be an option. Or it is what you already talked about some Future features being broken ?

Reply:
Yes it is. We would need to make followUser return a future which is inconvenient for people who want to use them in the app and then have to handle the futures that are only useful for testing.

Reply:
You can supply your test as an `CompletableFuture.supplyAsync` and join before the asserts. You do not need the function to return a future.